### DISCUSSION TITLE
Follow link when clicking on sub-menu

### DISCUSSION THREAD

**Discussion by jolajaslon** [vadikom/smartmenus#252] [Q&A] [answered]
I am testing SmartMenus version 2 instead of 1. In large-screen mode I want to have sub-menus that open automatically when I hover over them with the mouse, and if I remove the mouse they should close automatically. I've got this working great in version 2. However, when I click on a heading for a submenu in version 2 all that is done is that the submenu is closed. This is not what I want. When I click on a submenu heading with a link I want the browser to go to the link, not just hide the submenu. I have looked though the manual and the online configurator but I have not found a way to get SM2 to follow a submenu heading link when you click on it. I did try some hacking in the JS, and managed to get it partly working, but only partly... I am not much of a JavaScript hacker I'm afraid. Anyone have any ideas how I can get SM2 to behave the way I want and follow submenu heading links when you click on them?

**Comment by vadikom** [comment, marked as answer]
Hi, in v2 the script supports split links - i.e. menu items that have both a link part and a sub menu toggler button part - for example:

```html
<li class="sm-nav-item"><a class="sm-nav-link sm-nav-link--split" href="https://www.google.com">Google</a><button class="sm-nav-link sm-nav-link--split sm-sub-toggler"></button>
  <ul class="sm-sub">
    ...
  </ul>
</li>
```

which would result in:

![Screenshot from 2024-09-23 16-41-47](https://github.com/user-attachments/assets/4d97e0a0-7f32-45a5-8c00-9eaf63fa651e)

Compared to a regular parent item that only acts as a sub menu toggler button on click - e.g.:

```html
<li class="sm-nav-item"><a class="sm-nav-link" href="#">Sub</a>
  <ul class="sm-sub">
    ...
  </ul>
</li>
```

which looks like:

![Screenshot from 2024-09-23 16-57-48](https://github.com/user-attachments/assets/8427a21d-0318-432c-927d-dae8e7141db7)

You could still set an actual URL in the `href` attribute of the link for regular items but that would be accessible on click just for browsers with disabled JavaScript support.

So basically the idea in v2 is to use such split items whenever you need to set an actual link to your parent items.

Do you have any concerns about using such split items?

**Reply by jolajaslon** [reply]
This seems to work well. I wasn't aware that you had to add "--split" classes both to sm-nav-item and sm-sub-link a hrefs's and also add the button to get the links to work. I have tested it on my menu now and it works, I just need to get the layout of the buttons to look the way I want, but now I'm on the right track again. Thanks for your help!

Cheers'

Jonas
